### PR TITLE
feat: Adding to share button text twitter handle of generative artists

### LINKED ISSUE #9197: Adding to share button text twitter handle of generative artists
I think it would be a neat feature to help you tag artists that you collected from, especially for generative collections. 

old copy
> Check out this awesome NFT on KodaDot https://canary.kodadot.xyz/ahp/gallery/82-11 via @KodaDot 

new copy
> Check out this awesome NFT on @KodaDot from @buitrago91 https://canary.kodadot.xyz/ahp/gallery/82-11

### PR DESCRIPTION
**Pull request by Jarsen136**
**Thank you for your contribution** to the [KodaDot - One Stop Shop for Polkadot NFTs](https://kodadot.xyz).

  👇 __ Let's make a quick check before the contribution.

  ## PR Type
  - [x] Feature

  ## Needs QA check

  - @kodadot/qa-guild please review

  ## Context

  - [x] Closes #9197

  #### Did your issue had any of the "$" label on it?

  - [x] My DOT address: [Payout](https://canary.kodadot.xyz/dot/transfer/?target=16SjUbGKSdjCdWTy3NNT3JxbRVGGqD4mwkHpc6BD9U2Rp29Z)

  ## Screenshot 📸

  - [ ] My fix has changed UI

  ## Copilot Summary
  copilot:summary

  copilot:poem

### CODE REVIEW

**Comment by prury** [comment]
does not work for me?

image

**Comment by Jarsen136** [comment]
> does not work for me?
> 
> image

Interesting, It works on my local. I will check
image

**Comment by prury** [comment]
> ✅ It works. Could you pls check again. @prury

yeah, works now, but it seems it only works if the handle starts with @
check genera for example: https://deploy-preview-9223--koda-canary.netlify.app/ahp/gallery/63-435

File: utils/parse.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+export const extractTwitterIdFromDescription = (desc: string = '') => {
+  const regex =
+    /(?:https?:\/\/)?(?:www\.)?twitter\.com\/(?:\#!\/)?@?([^\/\?\s)]+)/
```
Comment (roiLeo) [inline utils/parse.ts:3]: please update regex to make `https://x.com/` valid

Comment (Jarsen136) [reply utils/parse.ts]: done

**Comment by Jarsen136** [comment]
> > ✅ It works. Could you pls check again. @prury
> 
> yeah, works now, but it seems it only works if the handle starts with @ check genera for example: https://deploy-preview-9223--koda-canary.netlify.app/ahp/gallery/63-435
> 
> edit: split also: https://deploy-preview-9223--koda-canary.netlify.app/ahp/gallery/66-252
> 
> nothing to do with the @ tho, some don't have it and work anyway

✅ Fixed

**Review by preschian — APPROVED** [review]

**Comment by yangwao** [comment]
Thanks!
pay 30 usd

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $30 @ 6.9 USD/DOT ~ 4.348 $DOT
🧗 [16SjUbGKSdjCdWTy3NNT3JxbRVGGqD4mwkHpc6BD9U2Rp29Z](https://polkadot.subscan.io/account/16SjUbGKSdjCdWTy3NNT3JxbRVGGqD4mwkHpc6BD9U2Rp29Z)
🔗 [0x6fa7532757b88d5b1200d605abb4e5992a3d708104feada5dd35f79e953f32c4](https://polkadot.subscan.io/extrinsic/0x6fa7532757b88d5b1200d605abb4e5992a3d708104feada5dd35f79e953f32c4)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Pull request merged**